### PR TITLE
docs: migration from native composition environment

### LINKED ISSUE #43: Clearly document migration steps from in-tree composition environment
### What problem are you facing?

The readme has an example of migration, but it would be good to have an explicit section about `Migration from --enable-environment-configs` or similar.

### How could this Function help solve your problem?

Update the README.

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Fixes #43
Follow up to https://github.com/crossplane/crossplane/pull/5938, already mentioning the `crossplane beta convert pipeline-composition` which still has to be implemented.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] ~Added or updated unit tests for my change.~

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,108 +6,206 @@
...
+
 This [composition function][docs-functions] allows you to request
 `EnvironmentConfigs`, merge them in the requested order and return inject the
 computed `environment` into the `Context` at a well-known key,
 `apiextensions.crossplane.io/environment`, so that other functions such as
 [function-patch-and-transform] can access it.
 
-The following paths from native `Resource` Compositions can be just moved to the
-input of this function:
+## Examples
```
Comment (tampakrap) [inline README.md:20]: This is now kinda duplicating the `## Using this function` section

File: README.md [hunk 2]
```diff
@@ -6,108 +6,206 @@
...
+`spec.environment.patches` and resources' `*Environment` patches will have to
+be moved to [function-patch-and-transform]'s input.
 
-Here's an example using [function-go-templating](https://github.com/crossplane-contrib/function-go-templating):
+The diagram below shows what part of the usual Composition is replaced by this
+function and how it fits with other functions:
+
+![diagram.png](diagram.png)
+
+## Migration Example
```
Comment (tampakrap) [inline README.md:50]:
```suggestion
### Migration Example
```

File: README.md [hunk 3]
```diff
@@ -151,9 +249,11 @@ $ docker build . --tag=runtime
 $ crossplane xpkg build -f package --embed-runtime-image=runtime
 ```
 
-[docs-functions]: https://docs.crossplane.io/v1.14/concepts/composition-functions/
 [bsr]: https://buf.build/crossplane/crossplane/docs/main:apiextensions.fn.proto.v1beta1#apiextensions.fn.proto.v1beta1.RunFunctionRequest
-[go]: https://go.dev
-[docker]: https://www.docker.com
 [cli]: https://docs.crossplane.io/latest/cli
+[docker]: https://www.docker.com
+[docs-functions]: https://docs.crossplane.io/v1.14/concepts/composition-functions/
```
Comment (tampakrap) [inline README.md:255]:
```suggestion
[docs-functions]: https://docs.crossplane.io/latest/concepts/composition-functions/
```

**Review by tampakrap — APPROVED** [review]
Very nice, thank you for providing it! I submitted some 💅 and minor suggestions. I have two extra suggestions:
- How about adding a ToC at the top
- Maybe to move the Migration section lower in the document, possibly between the Using and Developing sections

The content looks good though, thank you!

**Comment by tampakrap** [comment]
and another suggestion would be to put somewhere the link to the official doc https://docs.crossplane.io/latest/concepts/environment-configs/ I'm not sure what would be the best place, maybe an extra section or maybe just a phrase in the `Using this function` section?

File: README.md (hunk 1 above)
Comment (phisco) [reply README.md:20]: merged and moved that here.

File: README.md [hunk 4]
```diff
@@ -6,132 +6,227 @@
...
+  --include-full-xr \
+  example/xr.yaml example/composition.yaml example/functions.yaml | crossplane beta validate example -
+```
+
+See the [composition functions documentation][docs-functions] to learn more
+about `crossplane beta render`.
+
+## Migration from native Composition Environment
+
+Crossplane 1.8 [dropped] native Composition Environment, meaning
```
Comment (jbw976) [inline README.md:46]: should this be 1.18?

**Pull request merged**